core/dbt/main.py

```diff
@@ -143,7 +149,8 @@ def main(args=None):
 
         except BaseException as e:
             fire_event(MainEncounteredError(exc=str(e)))
-            fire_event(MainStackTrace(stack_trace=traceback.format_exc()))
+            if not isinstance(e, ParsingException):
```

Comment:
This would avoid generate the whole stack trace after the dbt command finish for Parsing Exception.